src/data_source/storage/sql.rs

```diff
@@ -1170,6 +1188,491 @@ where
     }
 }
 
+#[async_trait]
+impl<Types: NodeType, State: MerklizedState<Types>> UpdateStateData<Types, State> for SqlStorage
+where
+    State::Index: Serialize + Send + Sync + Display + FromStr + 'static,
+    State::Element: Serialize + Send + Sync + 'static,
+    State::NodeValue: Send + 'static,
```

Comment:
Should be able to remove this by adding trait bounds where the associated types are defined